src/app/CMakeLists.txt

```diff
@@ -1,162 +1,184 @@
-add_executable(qBittorrent
-    # headers
-    application.h
-    applicationinstancemanager.h
-    cmdoptions.h
-    filelogger.h
-    qtlocalpeer/qtlocalpeer.h
-    upgrade.h
-
-    # sources
-    application.cpp
-    applicationinstancemanager.cpp
-    cmdoptions.cpp
-    filelogger.cpp
-    main.cpp
-    qtlocalpeer/qtlocalpeer.cpp
-    upgrade.cpp
-)
+# Generate and configure translation files
+# -----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
+# Based on https://gist.github.com/giraldeau/546ba5512a74dfe9d8ea0862d66db412
+file(GLOB QBT_TS_FILES "${qBittorrent_SOURCE_DIR}/src/lang/*.ts")
+set_source_files_properties(${QBT_TS_FILES} PROPERTIES OUTPUT_LOCATION "${qBittorrent_BINARY_DIR}/src/lang")
+qt5_add_translation(QBT_QM_FILES ${QBT_TS_FILES})
+configure_file("${qBittorrent_SOURCE_DIR}/src/lang/lang.qrc" "${qBittorrent_BINARY_DIR}/src/lang/lang.qrc" COPYONLY)
 
-target_include_directories(qBittorrent PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
-target_link_libraries(qBittorrent PRIVATE qbt_base)
+if (WEBUI)
+    file(GLOB QBT_WEBUI_TS_FILES "${qBittorrent_SOURCE_DIR}/src/webui/www/translations/*.ts")
+    set_source_files_properties(${QBT_WEBUI_TS_FILES}
+        PROPERTIES OUTPUT_LOCATION "${qBittorrent_BINARY_DIR}/src/webui/www/translations")
+    qt5_add_translation(QBT_WEBUI_QM_FILES ${QBT_WEBUI_TS_FILES})
+    configure_file("${qBittorrent_SOURCE_DIR}/src/webui/www/translations/webui_translations.qrc"
+        "${qBittorrent_BINARY_DIR}/src/webui/www/translations/webui_translations.qrc" COPYONLY)
+endif()
 
-set_target_properties(qBittorrent
-    PROPERTIES
-        AUTOUIC True
-        AUTORCC True
-        MACOSX_BUNDLE True
-        RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}"
+FILE(GLOB QT_TRANSLATIONS "${qBittorrent_SOURCE_DIR}/dist/qt-translations/qtbase_*.qm")
+foreach(EXTRA_TRANSLATION IN ITEMS "fa" "gl" "lt" "pt" "sl" "sv" "zh_CN")
+    list(APPEND QT_TRANSLATIONS "${qBittorrent_SOURCE_DIR}/dist/qt-translations/qt_${EXTRA_TRANSLATION}.qm")
+endforeach()
+
+# Executable target configuration
+# -----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
+add_executable(qbt_app)
```

Comment:
@glassez I changed it for good reason - to not clash with the project name (the name passed in the call to `PROJECT()`). Otherwise it is confusing and there may be surprises, because CMake defines variables based on the project name.